clang/lib/Parse/ParseOpenMP.cpp

```diff
@@ -2918,6 +2950,29 @@ StmtResult Parser::ParseOpenMPDeclarativeOrExecutableDirective(
         ConsumeToken();
       Actions.EndOpenMPClause();
     }
+
+    // manually add num_threads clause if parallel directive doesn't have one in
+    // 'target thread_limit()' scope
+    if (clang::isOpenMPParallelDirective(DKind) &&
+        !FirstClauses[unsigned(OMPC_num_threads)].getInt()) {
+      OMPClause *num_threads_clause;
```

Comment:
Must initialize to NULL!
```suggestion
      OMPClause *num_threads_clause = nullptr;
```